relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -220,29 +325,111 @@ impl FieldValueProvider for Event {
         }
         RuleType::Error
     }
+    fn get_custom_operator(
+        name: &str,
+    ) -> fn(condition: &CustomCondition, slf: &Self, ip_addr: Option<IpAddr>) -> bool {
+        match name {
+            "event.client_ip" => client_ips_matcher,
+            "event.legacy_browser" => legacy_browsers_matcher,
+            "event.error_messages" => error_messages_matcher,
+            "event.csp" => csp_matcher,
+            _ => no_match,
+        }
+    }
+}
+
+fn client_ips_matcher(
+    condition: &CustomCondition,
+    _event: &Event,
+    ip_addr: Option<IpAddr>,
+) -> bool {
+    let ips = condition
+        .value
+        .as_array()
+        .map(|v| v.iter().map(|s| s.as_str().unwrap_or("")));
+
+    if let Some(ips) = ips {
+        client_ips::matches(ip_addr, ips)
+    } else {
+        false
+    }
+}
+
+fn legacy_browsers_matcher(
+    condition: &CustomCondition,
+    event: &Event,
+    _ip_addr: Option<IpAddr>,
+) -> bool {
+    let browsers = condition.value.as_array().map(|v| {
+        v.iter()
+            .map(|s| LegacyBrowser::from_str(s.as_str().unwrap_or("")).unwrap())
```

Comment:
_nit_: Thanks to `FromStr`, you can now use `parse`:

```suggestion
            .map(|s| s.as_str().unwrap_or("").parse().unwrap())
```